flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/IncrementalSourceReader.java

```diff
@@ -129,12 +136,27 @@ public List<SourceSplitBase> snapshotState(long checkpointId) {
         // add stream splits who are uncompleted
         stateSplits.addAll(uncompletedStreamSplits.values());
 
+        for (SourceSplitBase split : stateSplits) {
+            if (split.isStreamSplit()) {
+                lastCheckPointStreamSplit.put(checkpointId, split.asStreamSplit());
```

Comment:
> what about this? In for loop, if not just one meet the requirement, later one will override the later one in heap

The code has been moved to IncrementalSourceReaderWithCommit. Currently,  only StreamSplit is need, and should be just one StreamSplit for a checkpoint. Maybe a TreeMap<Long, StreamSplit> is enough, instead of TreeMap<Long, List<>>, do you think so?

Reply:
Thanks for your explain. It seems that TreeMap<Long, StreamSplit> are different of same streamSplit. Each streamSplit contains lots of redundant informations, such as tableSchemas and finishedSnapshotSplitInfos. It seems that only startingOffset is what we need. Why not change TreeMap<Long, StreamSplit> to TreeMap<Long, Offset>?

Reply:
Thanks and gree with you

Reply:
> Thanks for your explain. It seems that TreeMap<Long, StreamSplit> are different versions of same streamSplit. Each streamSplit contains lots of redundant informations, such as tableSchemas and finishedSnapshotSplitInfos. It seems that only startingOffset is what we need. Why not change TreeMap<Long, StreamSplit> to TreeMap<Long, Offset>?

Have made the modifications according to your suggestions. Could you please do a review if you have time?